### ISSUE TITLE
al_draw_soft_line shouldn't lock the bitmap by default

### ISSUE THREAD

**Issue by SiegeLord** [liballeg/allegro5#475]
This function is meant to be usable for non-drawing purposes, so it shouldn't lock the bitmap at all. For actual drawing use cases, the locking should be done in the `first` callback.

**Comment by beoran** [comment]
The problem I see is that while the bitmap can be locked by the user in the `first`callback, there is no `last` callback to unlock the bitmap in. Aslo, the locking used is not trivial, but based on the expected line that will be drawn. Which is actually wrong, since I could be drawing , say, large circles on a line and then the locked area on the bitmap will be too small. Furthermore, al _draw_soft_line is used by other software rendering functions,so I'm not too sure what will happen i we don't lock in that case. 

To deal with all this, I propose to add a function called al_do_soft_line that does no locking at all, and that will then be called by al_draw_soft_line.  al_dra_soft_line will then simply lock the whole bitmap.

**Comment by SiegeLord** [comment]
That's a very good point... perhaps maybe it should just check if the bitmap is NULL or not, before locking it.

**Comment by beoran** [comment]
Yes, NULL checking is actually a good idea.  A NULL bitmap will not be locked, a non-NULL one will. 
The user of the function can pass in a bitmap though uintptr_t state if they still need a pointer to it.
This is also a minimal invasive patch, so I'll probably make a PR for it if no one else does.
